src/v/kafka/server/handlers/produce.cc

```diff
@@ -165,13 +165,13 @@ static partition_produce_stages partition_append(
   model::partition_id id,
   ss::lw_shared_ptr<replicated_partition> partition,
   model::batch_identity bid,
-  model::record_batch batch,
+  std::unique_ptr<model::record_batch> batch,
```

Comment:
usage of unique_ptr in function signatures looks pretty artificial to me and constrains callers to use it as well. Can we achieve the same effect (reducing the number of expensive moves) by e.g. accepting ordinary or rvalue references?

Reply:
I know, it looks weird, this is strictly to avoid the moves. In here those function are local for the produce handler.  There is a transfer of the ownership. Maybe a comment would suffice ?